src/test/java/org/apache/commons/text/lookup/DnsStringLookupTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+
+package org.apache.commons.text.lookup;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests {@link DnsStringLookup}.
+ */
+public class DnsStringLookupTest {
+
+    @Test
+    public void testAddressFromHostAddress() throws UnknownHostException {
+        final InetAddress localHost = InetAddress.getLocalHost();
+        Assertions.assertEquals(localHost.getHostAddress(),
+                DnsStringLookup.INSTANCE.lookup("address|" + localHost.getHostAddress()));
```

Comment:
I see these tests as problematic because they require network connection or you rely on the fact that DNS entries are correct. You'd actually need a mock DNS server.

Reply:
A mock DNS server is over the top here IMO for what is essentially a JRE API facade. 

That said, I won't stock you from scratching that itch ;-) But then what about the rest of Commons?

Reply:
I know that this is over the top, but it would make the tests reliable.